extensions/notebook/src/jupyter/localPipPackageManageProvider.ts

```diff
@@ -0,0 +1,71 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { IPackageManageProvider, IPackageDetails, IPackageTarget } from '../types';
+import { JupyterServerInstallation } from './jupyterServerInstallation';
+import * as constants from '../common/constants';
+
+export class LocalPipPackageManageProvider implements IPackageManageProvider {
+
+	/**
+	 * Provider Id for Pip package manage provider
+	 */
+	public static ProviderId = 'localhost_Python';
+
+	constructor(private jupyterInstallation: JupyterServerInstallation) {
+	}
+
+	/**
+	 * Returns provider Id
+	 */
+	public get providerId(): string {
+		return LocalPipPackageManageProvider.ProviderId;
+	}
+
+	/**
+	 * Returns package target
+	 */
+	public get packageTarget(): IPackageTarget {
+		return { location: constants.localhostName, packageType: constants.PythonPkgType.Pip };
+	}
+
+	/**
+	 * Returns list of packages
+	 */
+	public async listPackages(): Promise<IPackageDetails[]> {
+		return await this.jupyterInstallation.getInstalledPipPackages();
+	}
+
+	/**
+	 * Installs give packages
```

Comment:
given 😄

Reply:
:((

---
In reply to: [350547959](https://github.com/microsoft/azuredatastudio/pull/8473#discussion_r350547959) [](ancestors = 350547959)